locked-nft/contracts/NFTLocker.cdc

```diff
@@ -172,6 +198,30 @@ pub contract NFTLocker {
             destroy oldToken
         }
 
+        pub fun extendLock(id: UInt64, nftType: Type, extendedDuration: UInt64) {
+            pre {
+                NFTLocker.nftIsLocked(
+                    id: id,
+                    nftType: nftType
+                ) == true : "token is not locked"
```

Comment:
I think there could be another validation for `old lockedUntil + extendedDuration > now` otherwise the extension seems moot